File: homework-05/src/main/java/ru/otus/homework/domain/Author.java
package ru.otus.homework.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

@Data
@AllArgsConstructor
@RequiredArgsConstructor
public class Author {
    private long id;

    @NonNull
    private String firstName;

    @NonNull
    private String lastName;
}

File: homework-11-ui/src/domain/Author.ts
export class Author {
    id: string;
    firstName: string;
    lastName: string;

    constructor(id: string, firstName: string, lastName: string) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
    }

    setId(value: string) {
        this.id = value;
    }

    setFirstName(value: string) {
        this.firstName = value;
    }

    setLastName(value: string) {
        this.lastName = value;
    }


}